Lib/test/test_itertools.py

```diff
@@ -15,6 +15,20 @@
 import struct
 import threading
 import gc
+import warnings
+
+def pickle_deprecated(testfunc):
+    """ Run the test twice. On the first pass, verify that a
```

Comment:
Is it worth it to run them a third time, with DeprecationWarnings promoted to errors? That usually ends up exercising paths that rarely get considered, but it'd probably be more in pickle than in itertools.